### ISSUE TITLE
Nested ordered list not starting at 1 broken

### ISSUE THREAD

**Issue by MofX** [commonmark/commonmark-spec#599]
The following sample renders the nested ordered list starting with 3. incorrectly
```
3. foo
2. foo
1. foo

------------------

* foo
  3. foo
  2. foo
  1. foo

------------------

* foo
  1. foo
  2. foo
  3. foo

------------------
```

See: [Sample](https://spec.commonmark.org/dingus/?text=3.%20foo%0A2.%20foo%0A1.%20foo%0A%0A------------------%0A%0A1.%20foo%0A2.%20foo%0A3.%20foo%0A%0A------------------%0A%0A*%20foo%0A%20%203.%20foo%0A%20%202.%20foo%0A%20%201.%20foo%0A%0A------------------%0A%0A*%20foo%0A%20%201.%20foo%0A%20%202.%20foo%0A%20%203.%20foo%0A%0A------------------)

**Comment by kivikakk** [comment]
Rendered for context:

* foo
  3. foo
  2. foo
  1. foo

Using the interactive dingus (what a name!) to [render this example](https://spec.commonmark.org/dingus/?text=*%20foo%0A%20%203.%20foo%0A%20%202.%20foo%0A%20%201.%20foo) shows the problem -- here's the AST:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">
<document xmlns="http://commonmark.org/xml/1.0">
  <list type="bullet" tight="true">
    <item>
      <paragraph>
        <text>foo</text>
        <softbreak />
        <text>3. foo</text>
        <softbreak />
        <text>2. foo</text>
      </paragraph>
      <list type="ordered" start="1" tight="true" delimiter="period">
        <item>
          <paragraph>
            <text>foo</text>
          </paragraph>
        </item>
      </list>
    </item>
  </list>
</document>
```

The `3.` is not recognised as the start of a list -- it is not separated from the previous line by a newline, and in §5.2 we note the exception:

> When the first list item in a list interrupts a paragraph—that is, when it starts on a line that would otherwise count as paragraph continuation text—then (a) the lines Ls must not begin with a blank line, and (b) if the list item is ordered, the start number must be 1.

The start number here is three, so it instead counts as paragraph continuation text. This repeats for the `2.` line, and then the condition is met for the last line, where the first actual ordered list is created. 

---

In other words, this is expected. If you insert a newline, like this:

```md
* foo

  3. foo
  2. foo
  1. foo
```

The following results:

* foo

  3. foo
  2. foo
  1. foo

Note the numbers _ascend_ from 3, instead of descending, hence "start number".

**Comment by MofX** [comment]
Looks like it really is according to spec, but still a bit weird. I guess this is just the nature of markdown...
